cxx-qt-gen/src/cxx_qt.h

```diff
@@ -73,53 +84,100 @@ class CxxQObject : public QObject
   // TODO: we need to document the existence of UpdateStateEvent for users who
   // want to create custom classes that derive from CxxQObject so that they
   // know to avoid clashes with it.
-  static const QEvent::Type UpdateStateEvent;
+  static const QEvent::Type ProcessQueueEvent;
 
 public:
   CxxQObject(QObject* parent = nullptr)
     : QObject(parent)
   {}
   virtual ~CxxQObject() = default;
 
-  void requestUpdate()
+  void requestEmitSignal(std::function<void()> signalFunctor)
```

Comment:
This could actually be renamed to something like: `runOnGUIThread`. Because that's pretty much what this does now, right?

Reply:
Maybe I'm missing something here, but why are all the functions implemented in the header file, and not a separate `.cpp` file?

Reply:
Renamed to `runOnGUIThread` and moved to .cpp file. I think we were wondering if they would be inlined or templated.